graphql_compiler/__init__.py

```diff
@@ -184,16 +188,16 @@ def get_graphql_schema_from_orientdb_schema_data(schema_data, class_to_field_typ
                                        (string -> {string -> GraphQLType}). Used to override the
                                        type of a field in the class where it's first defined and all
                                        the class's subclasses.
-        hidden_classes: optional set of strings, classes to not include in the GraphQL schema.
+        hidden_classes: set of strings, classes to not include in the GraphQL schema.
+                        By default we hide the base OrientDB vertex class to make the schema
+                        more syntactically sweet.
```

Comment:
It's still `optional set of strings`

Reply:
I think that the terminology that we've used for maybe None is optional.

Reply:
I mean to say that this field cannot be None.

Reply:
This is probably safe, but it's still better practice in Python to set it to `None` and then internally to the function check for `None` and set it to whatever the default value is. To see why, try making a function with a mutable default argument, then mutate it, and see what the default argument is on the next call.

Reply:
haha we were just talking about this

Reply:
I think having a frozenset in this specific case is better. 

Using the 'if is None" approach does not communicate as well to the reader that the default value of hidden_classes is {OrientDBBaseVertexClassName}. In this case, it's also unclear what happens when you modify the `hidden_classes` parameter. Do we still hide the OrientDBBaseVertex class name in addition to the hidden classes you specified? You have to look at the code to see.

Reply:
This is precisely what the docstring is for, however. All of these things should be made clear in the docstring (and probably tested, as well). Relying on the code to communicate this is bad practice -- what happens when the code is refactored?

Using a `frozenset` means that the code is currently correct. However, it also means that we have to continue to be mindful to not accidentally change the type back to `set`, because of the problem we discussed above. Going from `frozenset` to `set` is a much smaller change than going from `None` to `set`, and I am not confident I'll remember to always watch out for it in code reviews on this function in the future. Are you?

Reply:
Ok fair. Will fix.